monkey/monkey_island/cc/services/reporting/report.py

```diff
@@ -178,32 +180,53 @@ def get_exploited():
 
     @staticmethod
     def get_stolen_creds():
-        PASS_TYPE_DICT = {'password': 'Clear Password', 'lm_hash': 'LM hash', 'ntlm_hash': 'NTLM hash'}
         creds = []
+
+        # stolen creds from system info collectors
         for telem in mongo.db.telemetry.find(
                 {'telem_category': 'system_info', 'data.credentials': {'$exists': True}},
                 {'data.credentials': 1, 'monkey_guid': 1}
         ):
             monkey_creds = telem['data']['credentials']
-            if len(monkey_creds) == 0:
-                continue
-            origin = NodeService.get_monkey_by_guid(telem['monkey_guid'])['hostname']
-            for user in monkey_creds:
-                for pass_type in PASS_TYPE_DICT:
-                    if pass_type not in monkey_creds[user] or not monkey_creds[user][pass_type]:
-                        continue
-                    username = monkey_creds[user]['username'] if 'username' in monkey_creds[user] else user
-                    cred_row = \
-                        {
-                            'username': username,
-                            'type': PASS_TYPE_DICT[pass_type],
-                            'origin': origin
-                        }
-                    if cred_row not in creds:
-                        creds.append(cred_row)
+            formatted_creds = ReportService._format_creds_for_reporting(telem, monkey_creds)
+            if formatted_creds:
+                creds.extend(formatted_creds)
+
+        # stolen creds from exploiters
+        for telem in mongo.db.telemetry.find(
+                {'telem_category': 'exploit', 'data.info.credentials': {'$exists': True}},
+                {'data.info.credentials': 1, 'monkey_guid': 1}
+        ):
```

Comment:
This should be in a separate method. Since credentials are the same, regardless of their origin, there should be credential class (dataclass?).

Reply:
There should be a method called `get_all_credentials_for_report` which would call `get_credentials_from_exploit_telems` and `get_credentials_from_info_colector_telems` or something similar.

Reply:
Separating it into methods for now